hibernate-core/src/main/java/org/hibernate/metamodel/MappingMetamodel.java

```diff
@@ -167,7 +169,7 @@ default EntityPersister locateEntityPersister(Class<?> byClass) {
 	 *
 	 * @see #findCollectionDescriptor
 	 */
-	CollectionPersister getCollectionDescriptor(String role);
+	CollectionPersister getCollectionDescriptor(@Nullable String role);
```

Comment:
Unfortunately, i had to mark this as nullable due to `loadedPersister = factory.getRuntimeMetamodels().getMappingMetamodel().getCollectionDescriptor( role )`in CollectionEntry, where role is nullable, i assume due to a nullcheck at #103. If null shouldn't ever be null, then we should be able to remove that check, and go ahead without the @Nullable on role. wdyt?

Reply:
I could solve this because at the `getCollectionDescriptor( role )` callsite mentioned above, role really only can be null if the loadedPersister is null, and that's covered by a check, so i think a cast to nonnull is adequate here?